.github/workflows/v2-deploy-pr.yml

```
@@ -0,0 +1,131 @@
+name: V2 Deploy PR
+
+on:
+  workflow_run:
+    workflows:
+      - "V2 Build PR"
+    types:
+      - completed
+
+permissions:
+  actions: read
+  pull-requests: write
+  contents: read
+
+concurrency:
+  group: deploy-v2-demos-${{ github.event.workflow_run.head_branch }}
+  cancel-in-progress: true
+
+jobs:
+  # Step 1: Prepare the build context
+  prepare-build-context:
+    if: github.event.workflow_run.conclusion == 'success'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download Build Context
+        uses: XanaduAI/cloud-actions/download-github-workflow-artifact@main
+        with:
+          workflow_run_id: ${{ github.event.workflow_run.id }}
+          artifact_name_regex: '^pr_info$'
+          github_token: ${{ github.token }}
+
+      - name: Check if Build Context file exists
+        id: build_context
+        env:
+          context_artifact_file_name: pr_info.zip
+        run: |
+          if test -f "$context_artifact_file_name"; then
+            echo "result=$context_artifact_file_name" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Unpack Build Information
+        if: steps.build_context.outputs.result != ''
+        run: unzip ${{ steps.build_context.outputs.result }}
+
+      - name: Read Build Information
+        id: read_build_info
+        if: steps.build_context.outputs.result != ''
+        uses: actions/github-script@v6
```

May as well bump this to the latest version that uses Node 20
```suggestion
        uses: actions/github-script@v7
```